src/mlpack/tests/ann_visitor_test.cpp

```diff
@@ -86,3 +86,34 @@ TEST_CASE("WeightSizeVisitorTest", "[ANNVisitorTest]")
   REQUIRE(weightSize == randomSize * randomSize + randomSize);
 }
 
+/**
+ * Test that WeightSizeVisitor works properly for WeightNorm layer.
+ */
+TEST_CASE("WeightSizeVisitorTestForWeightNormLayer", "[ANNVisitorTest]")
+{
+  size_t randomSize = arma::randi(arma::distr_param(1, 100));
+
+  Linear<>* linear = new Linear<>(randomSize, randomSize);
+  LayerTypes<> weightNorm = new WeightNorm<>(linear);
+
+  size_t weightSize = boost::apply_visitor(WeightSizeVisitor(),
+                                           weightNorm);
+
+  REQUIRE(weightSize == randomSize * randomSize + randomSize + 1);
+}
+
+/**
+ * Test that WeightSizeVisitor works properly for LSTM layer.
+ */
+TEST_CASE("WeightSizeVisitorTestForLSTMLayer", "[ANNVisitorTest]")
+{
+  size_t randomSize = arma::randi(arma::distr_param(1, 100));
+
+  LayerTypes<> lstm = new LSTM<>(randomSize, randomSize);
+
+  size_t weightSize = boost::apply_visitor(WeightSizeVisitor(),
+                                           lstm);
+
+  REQUIRE(weightSize == 4 * randomSize * randomSize + 7 * randomSize +
+      4 * randomSize * randomSize);
```

Comment:
Following up on @rcurtin's comment here too, maybe we should compare it with `lstm.Parameters().n_elem`. Thanks.

Reply:
Yeah changed